xpsparser/parser.py

```diff
@@ -22,42 +22,580 @@
+        
+        self.default_axis_channel_id = 0
+        self.default_primary_channel_id = 1
+        
+        for key in self.kwarg_keys:
+            if key in kwargs.keys():
+                setattr(self, key, kwargs[key])
+            elif key == 'axis_id':
+                setattr(self, key, self.default_axis_channel_id)
+            elif key == 'primary_channel_id':
+                setattr(self, key, self.default_primary_channel_id) 
+
+        self.primary_spectrum_indicators = ['region', 'Region']
+        self.external_channel_indicators = ['external channel', 'External Channel']
+        
+        self.group_metadata_attribute_map = {
+            'Acquisition Date':'timestamp',
+            'Dwell Time':'dwell_time',
+            'Group':'group_name',
+            'Number of Scans':'n_scans',
+            'Region':'spectrum_region',
+            'Excitation Energy':'excitation_energy',
+            'Values/Curve':'n_values',
+            'Source':'source_label',            
+            }
+        
+        self.settings_attribute_map = {
+            'Analysis Method':'analysis_method',
+            'Analyzer Lens':'analyzer_lens',
+            'Analyzer Slit':'analyzer_slit',
+            'Detector Voltage':'detector_voltage',
+            'Eff. Workfunction':'workfunction',
+            'Scan Mode':'scan_mode',
+            }
+        
+        self.default_energy_unit = 'eV'
+        
+        self.known_channel_labels = {'Ring Current':'ring current',
+                                     'I_mirror':'mirror current', 
+                                     'Excitation Energy':'excitation energy',
+                                     'TEY':'total electron yield',
+                                     }
+        
+        self.known_channel_units = {'[mA]':'mA',
+                                    '[V]':'V', 
+                                    '[eV]':'eV',
+                                    }
+        
+        self.known_device_names = {'AMC Mono (TCP)':'monochromator',
+                                   'UE56/2-PGM1 (TCP)':'beamline',
+                                   'ARMIN-ADC3':'armin',
+                                   'Armin10':'armin',
+                                   }
+        
+        self.default_primary_device_name = 'Phoibos Hemispherical Analyzer'
+        self.default_axis_device_name = 'HSA 3500 plus'
+        
+        self.default_method_type = 'XPS'
+        
+        self.precision = 3
+        self.line_nr = 0
+
+    def parseFile(self, filepath, commentprefix = '#'):
+        """Parse the .xy file into a list of dictionaries.
+        
+        Parsed data is stored in the attribute 'self.data'.
+        """
+        self.data = []
+        self.file_contents = []
+        self.prefix = commentprefix
+        self.filepath = filepath
+        self._loadFile(filepath)
+        self.global_header = {}
+        """ There is a global header for the whole file. First parse that."""
```

Comment:
These kind things string should be comments not string literals. It is minor, but string literals are created and cost resources.